.github/workflows/reusable_checks_cpp.yml

```diff
@@ -0,0 +1,85 @@
+name: "C++ Tests on all platforms & compilers"
+
+on:
+  workflow_call:
+    inputs:
+      CONCURRENCY:
+        required: true
+        type: string
+      MINIMAL:
```

Comment:
nit: Maybe call this `FAST` to better indicate its intention, or flip it and call it `FULL` 🤷